### PR TITLE
Websocket error handling

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -181,6 +181,18 @@ lazy val `kademlia` = crossProject(JVMPlatform, JSPlatform)
...
 lazy val `kademlia-jvm` = `kademlia`.jvm
 
+lazy val `transport-grpc-monix` = project
+  .in(file("transport/grpc-monix"))
+  .settings(
+    commons,
+    grpc,
+    libraryDependencies ++= Seq(
+      monix3,
+      "org.scalatest" %% "scalatest"      % ScalatestV % Test
```
Comment (RicoGit) [inline build.sbt]: There is a extra spaces here and we have a variable `scalatestKit` for this dependency.

File: node/src/main/scala/fluence/node/FluenceNode.scala [hunk 2]
```diff
@@ -137,7 +137,7 @@ object FluenceNode extends slogging.LazyLogging {
       fs2SchedulerWithShutdownTask ← fs2.Scheduler.allocate[IO](2)
       (fs2Scheduler, fs2Shutdown) = fs2SchedulerWithShutdownTask
       websocketServerShutdown ← GrpcWebsocketProxy
-        .startWebsocketServer(inProcessGrpc, fs2Scheduler, port)
+        .startWebsocketServer(inProcessGrpc, fs2Scheduler, port)(Scheduler.global)
```
Comment (RicoGit) [inline node/src/main/scala/fluence/node/FluenceNode.scala]: There is so many `Scheduler.global` in this file, let's create field for scheduler.

File: transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala [hunk 3]
```diff
@@ -50,21 +50,34 @@ object GrpcMonix {
       }
     }
 
-  implicit def streamToObserver[T](stream: StreamObserver[T])(implicit sch: Scheduler): Observer[T] = new Observer[T] {
-    override def onError(ex: Throwable): Unit =
-      stream.onError(ex)
+  implicit def streamToObserver[T](stream: StreamObserver[T])(implicit sch: Scheduler): Observer.Sync[T] =
```
Comment (RicoGit) [inline transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala]: Is there some special condition for making Sync Observer ? What if will be thrown exception?

File: transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala [hunk 4]
```diff
@@ -50,21 +50,34 @@ object GrpcMonix {
...
+
+      override def onNext(elem: T): Ack = {
+        Try {
+          stream.onNext(elem)
+          Ack.Continue
+        }.recover {
+          case e: Throwable ⇒
+            onError(e)
+            Ack.Stop
+        }.getOrElse(Ack.Stop)
```
Comment (RicoGit) [inline transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala]: Looks like `recover` is useless, can we invoke `onError(e)` into `getOrElse` block?
What happened when `onError` throw exception, I would write a test for this case.

File: transport/grpc-proxy/src/main/scala/fluence/grpc/proxy/GrpcWebsocketProxy.scala [hunk 5]
```diff
@@ -45,38 +42,38 @@ object GrpcWebsocketProxy extends Http4sDsl[Task] with slogging.LazyLogging {
...
             message ← Task(WebsocketMessage.parseFrom(data))
             _ = logger.debug(s"Handle websocket message $message")
+            // TODO message.response.payload.get.newInput() rewrite with error handling
             responseObservable ← proxyGrpc
-              .handleMessage(message.service, message.method, message.requestId, message.payload.newInput())
+              .handleMessage(
+                message.service,
+                message.method,
+                message.requestId,
+                message.response.payload.get.newInput()
```
Comment (RicoGit) [inline transport/grpc-proxy/src/main/scala/fluence/grpc/proxy/GrpcWebsocketProxy.scala]: Is it safe?

File: transport/websocket-js/src/main/scala/fluence/transport/websocket/StatusException.scala [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+package fluence.transport.websocket
+
+import fluence.proxy.grpc.Status
+
+/**
+ * Mirror of grpc.StatusException for js
+ * @param status Status code and description of exception.
+ */
+class StatusException(status: Status) extends Throwable {}
```
Comment (RicoGit) [inline transport/websocket-js/src/main/scala/fluence/transport/websocket/StatusException.scala]: There is extra brackets and empty line  is missing.

File: transport/grpc-proxy/src/main/scala/fluence/grpc/proxy/GrpcWebsocketProxy.scala (hunk 5 above)
Comment (DieMyst) [reply transport/grpc-proxy/src/main/scala/fluence/grpc/proxy/GrpcWebsocketProxy.scala]: no, i missed it

File: transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala (hunk 3 above)
Comment (DieMyst) [reply transport/grpc-monix/src/main/scala/fluence/grpc/GrpcMonix.scala]: It is easier to implement grpc converters with sync observer.
If there will be an exception, onError will be called

**Pull request merged**